dart/dynamics/Shape.h

```diff
@@ -191,29 +142,19 @@ class Shape : public virtual common::Subject
   /// Instruct this shape to update its data
   virtual void refreshData();
 
-  /// \brief
-  virtual void draw(renderer::RenderInterface* _ri = nullptr,
-                    const Eigen::Vector4d& _color = Eigen::Vector4d::Ones(),
-                    bool _useDefaultColor = true) const = 0;
-
-  /// Pass in true to prevent this shape from being rendered. Pass in false to
-  /// allow it to render again.
-  void setHidden(bool _hide=true);
+  /// Notify that the color (rgba) of this shape has updated
+  virtual void notifyColorUpdate(const Eigen::Vector4d& color);
```

Comment:
I believe DART already has that structure. You may have more than one `ShapeNode`s that reference a single `Shape` instance by `std::shared_ptr`. We do not have a central repository of all `Shape` instances, but it seems relatively straightforward for the viewer to construct that itself.

The only exception is `VisualAddOn`, which contains color, is attached to `ShapeNode` instead of `Shape`. One motivation of this is to avoid copying an entire mesh just to change RGBA values. Are you saying that this makes rendering inefficient because you can't render those `Shape`s in a batch?

I'm also still not sure what the API for textures should look like. The [Assimp materials API](http://assimp.sourceforge.net/lib_html/materials.html) is quite complicated.